### PR TITLE
[#8] Delete category created.

### LINKED ISSUE #8: Delete categories
As a user, I want to be able to **delete** my category.

### PR DESCRIPTION
**Pull request by sandeepsingh8**
closes #8

### CODE REVIEW

**Review by satgi — CHANGES_REQUESTED** [review]
1. No issue is linked to this pull request, as a result, the issue will not be closed automatically once it's merged.
2. According to the [issue description](https://github.com/KarlReyes28/Expired/issues/8#issuecomment-1401004837), the following features are not implemented yet.

- [x] A category can only be deleted if no product is linked.
- [ ] A confirmation alert will show to ensure the user is sure of their action.

File: Expired/Views/CategoriesView.swift [hunk 1]
```diff
@@ -10,18 +10,26 @@ import SwiftUI
 struct CategoriesView: View {
     @Environment(\.managedObjectContext) private var viewContext
     @EnvironmentObject var productStore: ProductStore
-
-    @State private var showingFormAlert: Bool = false
-    @State private var category: String = ""
+    @State private var showingNotDeleteAlert: Bool = false
+//    @State private var category: String = ""
```
Comment (donghua-shen) [inline Expired/Views/CategoriesView.swift:14]:
```suggestion

```

Delete unused variable

File: Expired/Views/CategoriesView.swift [hunk 2]
```diff
@@ -43,6 +68,22 @@ struct CategoriesView: View {
...
+            indexSet.map{productStore.categories[$0]}.forEach{ category in
+                if(category.products?.count == 0){
+                    viewContext.delete(category)
+                }else{
+                   showingNotDeleteAlert =  true
+                }
+                productStore.save(viewContext)
+            }
+        }
+    }
```
Comment (donghua-shen) [inline Expired/Views/CategoriesView.swift:82]:
```suggestion

    private func deleteCategory(indexSet: IndexSet) {
        withAnimation {
            indexSet.map{productStore.categories[$0]}.forEach{ category in
                if category.products?.count ?? 0 > 0 {
                    showingNotDeleteAlert = true
                } else {
                   viewContext.delete(category)
                }
            }
            productStore.save(viewContext)
        }
    }

```

- space matters
- Avoid invoking `productStore.save` multiple times by moving it outside of the loop

File: Expired/Views/CategoriesView.swift [hunk 3]
```diff
@@ -31,6 +39,23 @@ struct CategoriesView: View {
...
+                }
+                Button("Yes", role: .destructive) {
+                    if let indexSet = deleteIndexSet {
+                        deleteCategory(indexSet: indexSet)
+                    }
+                    deleteIndexSet = nil
+                }
+            }
+            .alert("Unable To Delete", isPresented: $showingNotDeleteAlert , actions: {
+                Button("Ok", role: .cancel) {
```
Comment (donghua-shen) [inline Expired/Views/CategoriesView.swift:54]:
```suggestion
                Button("OK", role: .cancel) {
```

- uppercase the "OK" text

**Review by donghua-shen — CHANGES_REQUESTED** [review]
Reviewed, thanks! 👍

**Review by donghua-shen — APPROVED** [review]

**Pull request merged**